src/runtime/exceptions.cpp

```diff
@@ -203,6 +203,8 @@ extern "C" void raise3(STOLEN(Box*) arg0, STOLEN(Box*) arg1, STOLEN(Box*) arg2)
         startReraise();
 
     assert(!PyErr_Occurred());
+    if (PyErr_Occurred())
```

Comment:
These lines seem very odd when combined with the assert above them :)  Maybe we should just get rid of the assert if we can support this case reasonably?